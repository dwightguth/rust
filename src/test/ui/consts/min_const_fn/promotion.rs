#![feature(min_const_fn)]

const fn foo1() {}
const fn foo2(x: i32) -> i32 { x }

fn main() {
    let x: &'static () = &foo1();
    let y: &'static i32 = &foo2(42); //~ ERROR does not live long enough
}