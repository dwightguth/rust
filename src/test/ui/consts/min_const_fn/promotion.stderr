error[E0597]: borrowed value does not live long enough
  --> $DIR/promotion.rs:8:28
   |
LL |     let y: &'static i32 = &foo2(42); //~ ERROR does not live long enough
   |                            ^^^^^^^^ temporary value does not live long enough
LL | }
   | - temporary value only lives until here
   |
   = note: borrowed value must be valid for the static lifetime...

error: aborting due to previous error

For more information about this error, try `rustc --explain E0597`.
